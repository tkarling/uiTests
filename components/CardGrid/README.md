# @ui-tests/card-grid

Welcome to my `CardGrid` component.

## Installation

```sh
npm i @ui-tests/card-grid
# or with yarn
yarn add @ui-tests/card-grid
```

Then to use the component in your code just import it!

```js
import CardGrid from '@ui-tests/card-grid';
// and with css-modules
import '@ui-tests/card-grid/dist/main.css';
```
