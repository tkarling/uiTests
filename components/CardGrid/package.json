{
  "name": "@ui-tests/card-grid",
  "version": "0.0.0",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/index.js",
  "types": "./dist",
  "repository": "tkarling",
  "author": "tkarling tkarling@live.com",
  "contributors": [
    "tkarling tkarling@live.com"
  ],
  "license": "MIT",
  "sideEffects": [
    "*.css"
  ],
  "scripts": {
    "clean": "ds clean",
    "build": "ds build",
    "dev": "ds dev",
    "storybook": "ds storybook start",
    "start": "ds build --watch",
    "test": "ds test",
    "lint": "ds lint",
    "size": "ds size"
  },
  "dependencies": {
    "@babel/runtime": "7.7.2"
  },
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist",
    "src",
    "!*.snippet.*",
    "!*theme.*",
    "!*.test.*",
    "!__snapshots__",
    "!__tests__"
  ]
}
