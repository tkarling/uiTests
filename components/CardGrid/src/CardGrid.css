.page {
  border: 1px solid green;
  padding: 10px;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen,
    Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
}
.pageTitle {
  padding-bottom: 10px;
  font-weight: bold;
  font-size: large;
}
/* .card-grid-xs {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
} */

.card-grid {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}

.card-xs {
  border: 1px solid purple;
  padding: 10px;
  width: 100%;
  margin-bottom: 10px;
}

.card {
  border: 1px solid blue;
  padding: 10px;
  width: 30%;
  margin-bottom: 10px;
}
