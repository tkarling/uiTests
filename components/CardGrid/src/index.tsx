import React from 'react';
import styles from './CardGrid.css';
import useMedia, { MediaSize } from './useMedia';

const Card = ({
  children,
  media
}: {
  children: Node | string;
  media: MediaSize;
}) => {
  const cardStyle = media === 'xs' ? styles.cardXs : styles.card;
  return <div className={cardStyle}>{children}</div>;
};

// const CARDS = ['moi', 'hei', 'joo', 'lii', 'laa', 'lee', 'loo'];
interface CardGridProps {
  cards?: string[];
}

export const CardGrid = ({ cards = [] }: CardGridProps) => {
  const { media } = useMedia();
  return (
    <div className={styles.page}>
      <div className={styles.pageTitle}>Number of Cards: {cards.length}</div>
      <div className={styles.cardGrid}>
        {cards.map(card => (
          <Card key={card} media={media}>
            {card}
          </Card>
        ))}
      </div>
    </div>
  );
};

export default CardGrid;
