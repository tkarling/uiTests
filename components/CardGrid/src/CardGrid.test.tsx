import React from 'react';
import { render } from '@testing-library/react';
import CardGrid from '.';

describe('CardGrid', () => {
  test('It matches the snapshot', () => {
    const { container } = render(<CardGrid />);
    expect(container).toMatchSnapshot();
  });
});
