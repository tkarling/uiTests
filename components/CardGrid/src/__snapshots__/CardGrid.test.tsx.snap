// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardGrid It matches the snapshot 1`] = `
<div>
  <div
    class="page"
  >
    <div
      class="pageTitle"
    >
      Number of Cards: 
      0
    </div>
    <div
      class="cardGrid"
    />
  </div>
</div>
`;
